zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/IndexSchemaValidator.java

```diff
@@ -0,0 +1,170 @@
+        LOGGER.info(message);
+
+      } else if (!difference.getEntriesOnlyOnLeft().isEmpty()) {
+        // Collect the new fields
+        newFields.put(indexDescriptor, difference.getEntriesOnlyOnLeft());
+      }
+    } else {
+      LOGGER.debug(
+          String.format(
+              "Index fields are up to date. Index name: %s.", indexDescriptor.getIndexName()));
+    }
+  }
+
+  private IndexMappingDifference getIndexMappingDifference(
+      final IndexDescriptor indexDescriptor, final Map<String, IndexMapping> indexMappingsGroup) {
+    final IndexMapping indexMappingMustBe = schemaManager.readIndex(indexDescriptor);
+
+    IndexMappingDifference difference = null;
+    // compare every index in group
+    for (final Map.Entry<String, IndexMapping> singleIndexMapping : indexMappingsGroup.entrySet()) {
+      final IndexMappingDifference currentDifference =
+          new IndexMappingDifference.IndexMappingDifferenceBuilder()
+              .setLeft(indexMappingMustBe)
+              .setRight(singleIndexMapping.getValue())
+              .build();
+      if (!currentDifference.isEqual()) {
+        if (difference == null) {
+          difference = currentDifference;
+          // If there is a difference between the template and the existing runtime/data
+          //     indices,
+          // all those indices should have the same difference. Compare based only on the
+          // differences (exclude the IndexMapping fields in the comparison)
+        } else if (!difference.checkEqualityForDifferences(currentDifference)) {
+          throw new IndexSchemaValidationException(
+              "Ambiguous schema update. First bring runtime and date indices to one schema. Difference 1: "
+                  + difference
+                  + ". Difference 2: "
+                  + currentDifference);
+        }
+      }
+    }
+    return difference;
+  }
+
+  //
+  /**
+   * Leave only runtime and dated indices that correspond to the given IndexDescriptor.
+   *
+   * @param indexMappings
+   * @param indexDescriptor
+   * @return
+   */
+  private Map<String, IndexMapping> filterIndexMappings(
+      final Map<String, IndexMapping> indexMappings, final IndexDescriptor indexDescriptor) {
+    if (indexDescriptor instanceof IndexTemplateDescriptor) {
+      return Maps.filterEntries(
+          indexMappings,
+          e -> e.getKey().equals(((IndexTemplateDescriptor) indexDescriptor).getTemplateName()));
+    } else {
+      return Maps.filterEntries(
+          indexMappings,
+          e -> e.getKey().matches(indexDescriptor.getAllVersionsIndexNameRegexPattern()));
+    }
+  }
+
+  private boolean indexIsDynamic(final IndexMapping mapping) {
+    if (mapping == null) {
+      return false;
+    }
+    if (mapping.dynamic() == null) {
+      return true;
+    }
+
+    return Boolean.parseBoolean(mapping.dynamic());
+  }
+
+  private void validateFieldsDifferBetweenIndices(
```

Comment:
:wrench: Can be renamed to make the intention more clear. Something like `failIfIndexNotDynamic`